internal/archive/tar/tar.go

```diff
@@ -0,0 +1,70 @@
+package tar
+
+import (
+	"archive/tar"
+	"bytes"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/saucelabs/saucectl/internal/sauceignore"
+)
+
+// Resource archives the resource and exclude files and folders based on sauceignore logic.
+func Resource(src string, matcher sauceignore.Matcher) (io.Reader, error) {
+	bb := new(bytes.Buffer)
+	w := tar.NewWriter(bb)
+	defer w.Close()
+
+	walker := func(file string, fileInfo os.FileInfo, err error) error {
+		// Only will be applied if we have .sauceignore file and have patterns to exclude files and folders
+		if matcher.Match(strings.Split(file, string(os.PathSeparator)), fileInfo.IsDir()) {
+			return nil
+		}
+
+		header, err := tar.FileInfoHeader(fileInfo, file)
+		if err != nil {
+			return err
+		}
+
+		relFilePath := file
+		if filepath.IsAbs(src) {
+			// copy temp files
+			if strings.Contains(src, os.TempDir()) {
+				relFilePath = filepath.Base(file)
+			} else {
+				relFilePath, err = filepath.Rel(src, file)
+				if err != nil {
+					return err
+				}
+			}
+		}
+
+		header.Name = relFilePath
+
+		if err := w.WriteHeader(header); err != nil {
+			return err
+		}
+
+		if fileInfo.Mode().IsDir() {
```

Comment:
You are using `fileInfo.IsDir()` on https://github.com/saucelabs/saucectl/pull/235/files#diff-5f57135b1a4689f06fcac284a4639cbe5a54af21aa14041b8962b29e2e15df11R22

I know that the upper one is just syntactic suger, but would be good to at least be consistent within the same method 💡